airflow/providers/amazon/aws/waiters/batch.json

```diff
@@ -0,0 +1,19 @@
+{
+  "version": 2,
+  "waiters": {
+    "JobComplete": {
+      "delay": 300,
+      "operation": "DescribeJobs",
+      "maxAttempts": 100,
+      "description": "Wait until job is SUCCEEDED",
+      "acceptors": [
+        {
+          "argument": "jobs[].status",
+          "expected": "SUCCEEDED",
+          "matcher": "pathAll",
+          "state": "success"
+        }
```

Comment:
Similar to https://github.com/apache/airflow/pull/30945#discussion_r1183679249, can we add another acceptor for the `FAILED` state?

Reply:
Added